client/src/modules/stock/adjustment/adjustment.js

```diff
@@ -127,12 +122,32 @@ function StockAdjustmentController(Depots, Inventory, Notify,
       .catch(Notify.handleError);
   }
 
-  // init actions
-  function moduleInit() {
-    vm.movement = { date: new Date(), entity: {} };
-    loadInventories(vm.depot);
-    setupDepot(vm.depot);
-    checkValidity();
+  function setupStock() {
+    vm.Stock.setup();
+    vm.Stock.store.clear();
+  }
+
+  function startup() {
+    vm.movement = {
+      date : new Date(),
+      entity : {},
+    };
+
+    vm.depot = cache.depot;
+
+    // make sure that the depot is loaded if it doesn't exist at startup.
+    if (vm.depot) {
+      setupStock();
+      loadInventories(vm.depot);
```

Comment:
Technically, this repeated code could be combined into a single function:
```js
function setupStock(depot) {
  // old code
  loadInventories(vm.depot);
  checkValidity();
}

// later..

if (vm.depot) {
  setupStock(vm.depot);
} else {
  changeDepot()
    .then(setupStock);
}
```

Slight code reduction here by about two lines.